pkg/piperenv/artifact.go

```diff
@@ -1,7 +1,8 @@
 package piperenv
 
 type Artifact struct {
-	Name string `json:"name,omitempty"`
+	Name         string `json:"name,omitempty"`
+	PublishedUrl string `json:"publishedUrl,omitempty"`
```

Comment:
As this is going to be the standard all artifact producing steps should follow, what do you think about an more formal process (let's call it PEP for Piper Enhancement Proposal - following KEP for Kubernetes, GEP for Gardener and others) and talk about and decide on what we could/should standardize on.

cc @phil9909 @anilkeshav27 @CCFenner @OliverNocon